src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -230,8 +230,27 @@ public ReadOnlyMemory<byte> GetMessageBytes(HubMessage message)
                             else
                             {
                                 // If we have an invocation id already we can parse the end result
-                                var returnType = binder.GetReturnType(invocationId);
-                                result = BindType(ref reader, input, returnType);
+                                Type? returnType;
+                                try
+                                {
+                                    returnType = binder.GetReturnType(invocationId);
+                                }
+                                // GetReturnType throws if invocationId not found, this can be caused by the server canceling a client-result but the client still sending a result
+                                // For now let's ignore the failure and skip parsing the result, server will log that the result wasn't expected anymore and ignore the message
+                                // In the future we may want a CompletionBindingFailureMessage that we can flow to the dispatcher for handling
+                                catch (Exception)
+                                {
+                                    returnType = null;
+                                }
```

Comment:
Nit: Add a central TryGetReturnType helper to avoid duplicating this bit of code